contrib/cindercompatibleapi/converter/volume.go

```diff
@@ -0,0 +1,380 @@
+	ServiceUuID         string            `json:"service_uuid,omitempty"`
+	SharedTargets       bool              `json:"shared_targets,omitempty"`
+}
+
+// ShowVolumeResp ...
+func ShowVolumeResp(volume *model.VolumeSpec) *ShowVolumeRespSpec {
+	resp := ShowVolumeRespSpec{}
+
+	resp.Volume.Attachments = make([]RespAttachment, 0, 0)
+	resp.Volume.AvailabilityZone = volume.AvailabilityZone
+	resp.Volume.UpdatedAt = volume.BaseModel.UpdatedAt
+	resp.Volume.ID = volume.BaseModel.Id
+	resp.Volume.Size = volume.Size
+	resp.Volume.UserID = volume.UserId
+	resp.Volume.Metadata = make(map[string]string)
+	resp.Volume.Status = volume.Status
+	resp.Volume.Description = volume.Description
+	resp.Volume.Name = volume.Name
+	resp.Volume.CreatedAt = volume.BaseModel.CreatedAt
+	//resp.Volume.TenantID = volume.TenantId
+
+	return &resp
+}
+
+// *******************Update a volume*******************
+
+// UpdateVolumeReqSpec ...
+type UpdateVolumeReqSpec struct {
+	Volume UpdateReqVolume `json:"volume"`
+}
+
+// UpdateReqVolume ...
+type UpdateReqVolume struct {
+	Description string            `json:"description,omitempty"`
+	Name        string            `json:"name,omitempty"`
+	Metadata    map[string]string `json:"metadata,omitempty"`
+}
+
+// UpdateVolumeRespSpec ...
+type UpdateVolumeRespSpec struct {
+	Volume UpdateRespVolume `json:"volume"`
+}
+
+// UpdateRespVolume ...
+type UpdateRespVolume struct {
+	MigrationStatus    string            `json:"migration_status,omitempty"`
+	Attachments        []RespAttachment  `json:"attachments"`
+	Links              []Link            `json:"links,omitempty"`
+	AvailabilityZone   string            `json:"availability_zone,omitempty"`
+	Encrypted          bool              `json:"encrypted ,omitempty"`
+	UpdatedAt          string            `json:"updated_at"`
+	ReplicationStatus  string            `json:"replication_status,omitempty"`
+	SnapshotID         string            `json:"snapshot_id,omitempty"`
+	ID                 string            `json:"id"`
+	Size               int64             `json:"size"`
+	UserID             string            `json:"user_id"`
+	Metadata           map[string]string `json:"metadata"`
+	Status             string            `json:"status"`
+	Description        string            `json:"description"`
+	Multiattach        bool              `json:"multiattach,omitempty"`
+	SourceVolID        string            `json:"source_volid,omitempty"`
+	ConsistencygroupID string            `json:"consistencygroup_id,omitempty"`
+	Name               string            `json:"name"`
+	Bootable           bool              `json:"bootable,omitempty"`
+	CreatedAt          string            `json:"created_at"`
+	VolumeType         string            `json:"volume_type,omitempty"`
+}
+
+// RespAttachment ...
+type RespAttachment struct {
+	ID string `json:"id,omitempty"`
+}
+
+// UpdateVolumeReq ...
+func UpdateVolumeReq(cinderReq *UpdateVolumeReqSpec) (*model.VolumeSpec, error) {
```

Comment:
I don't think this method should return error, since other methods only return ***RespSpec. You can package the validate operation in a standalone method.